lib/utils/HDFDatabase.h

```diff
@@ -227,19 +231,129 @@ class HDFDatabase : public Database
         int block_size,
         int stride);
 
-private:
     /**
-     * @brief Unimplemented copy constructor.
+     * @brief Creates a new HDF5 database file with the supplied name,
+     *        extending it with 6 digits indicating the process rank.
+     *
+     * @param[in] file_name Name of HDF5 database file to create.
+     * @param[in] comm      MPI communicator to obtain the process rank.
+     *
+     * @return True if file create was successful.
      */
-    HDFDatabase(
-        const HDFDatabase& other);
+    virtual bool
+    create_parallel(
```

Comment:
Would the interface be simpler if we just add optional inputs to the functions `create`, `open`, `close`, rather than adding new function names ending in `_parallel`? If so, maybe the `_parallel` functions could be kept as private functions and called by the functions without `_parallel` in the parallel case.

Reply:
We'd probably have to put the same optional input for `put...Array` and `get...Array` functions. I agree this would be a simpler interface to users in general.